frontend/packages/citizen-frontend/src/applications/editor/contact-info/OtherChildrenSubSection.tsx

```diff
@@ -64,15 +64,26 @@ export default React.memo(function OtherChildrenSubSection({
           updateFormData({
             otherChildrenExists: checked
           })
-          if (formData.otherChildren.length === 0) {
+          if (checked) {
+            if (formData.otherChildren.length === 0) {
+              updateFormData({
+                otherChildren: [
+                  {
+                    firstName: '',
+                    lastName: '',
+                    socialSecurityNumber: ''
+                  }
+                ]
+              })
+            }
+          } else {
             updateFormData({
-              otherChildren: [
-                {
-                  firstName: '',
-                  lastName: '',
-                  socialSecurityNumber: ''
-                }
-              ]
+              otherChildren: formData.otherChildren.filter(
+                (child) =>
+                  child.firstName.length > 0 ||
+                  child.lastName.length > 0 ||
+                  child.socialSecurityNumber.length > 0
```

Comment:
If you fill one of these fields and then uncheck the same issue will still happen? Should there be `&&` instead of `||`?

Reply:
Although still there would be case where you could fill in invalid ssn and have it fail while hidden. So actually the solution is not to remove rows when unchhecking, but rather edit the validation function so that it only validates those children if `otherChildrenExists === true`. That is one of the reasons why the checkbox boolean is part of the FormData even though it is not sent to backend.

Reply:
Thanks for the feedback @Joosakur. I'll make changes accordingly.